src/machine/machine_avr.go

```diff
@@ -157,3 +159,119 @@ func (a ADC) Get() uint16 {
 	high := uint16(*avr.ADCH)
 	return uint16(low) | uint16(high<<8)
 }
+
+// I2C on the Arduino
+type I2C struct {
+	// Cannot have just an empty struct or compile optimization fails.
+	Bus uint8
```

Comment:
Ouch, this is one reason why I was adding tests yesterday - I was afraid something would go wrong with empty structs. Ideally this should be fixed before merging. It's a good workaround but should not have been necessary.